pkg/accountmanager/manager.go

```diff
@@ -0,0 +1,74 @@
+package accountmanager
+
+import (
+	"fmt"
+
+	"github.com/Optum/dce/pkg/errors"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/arn"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/sts"
+	"github.com/aws/aws-sdk-go/service/sts/stsiface"
+)
+
+// AccountManager manages account resources
+type AccountManager struct {
+	awsConfig           *aws.Config
+	sts                 stsiface.STSAPI
+	adminRoleArn        arn.ARN
+	principalRoleName   string `env:"PRINCIPAL_ROLE_NAME"`
+	principalPolicyName string `env:"PRINCIPAL_POLICY_NAME"`
+	principalRoleArn    arn.ARN
+	principalPolicyArn  arn.ARN
+}
+
+// Setup creates a new session manager struct
+func (am *AccountManager) Setup(adminRoleArn string) error {
+	var err error
+
+	am.adminRoleArn, err = arn.Parse(adminRoleArn)
+	if err != nil {
+		return errors.NewValidation("admin role arn", err)
+	}
+
+	// Create the credentials from AssumeRoleProvider to assume the role
+	creds := stscreds.NewCredentialsWithClient(am.sts, am.adminRoleArn.String())
+	_, err = creds.Get()
+	if err != nil {
+		return errors.NewValidation("admin role arn", err)
+	}
+	am.awsConfig = aws.NewConfig().WithCredentials(creds)
+
+	am.principalRoleArn, err = arn.Parse(
```

Comment:
We're making some assumptions here that may not be safe to make.
We actually store the principal role ARN in the Account DynDB table. I could envision some weird bugs popping up here, if the `PRINCIPAL_ROLE_NAME` env var doesn't match whats in the account's DB record

Reply:
Yea, funny because we probably have that problem all over the place now too.  I think we need to decide how this gets set.  

The environment variable name should be the Default value only?

Reply:
> The environment variable name should be the Default value only?

Yeah, I'd guess I'd say the env var is used in the create account controller, to decide how to name the principal role/policy in the first place. But then it's persisted to the DB, and should probably be read from the DB in the future.

Or maybe we decide that persisting the role names to the DB is not useful, and drop it. But dong both is not great.

We can punt on this for now, if pulling these from the DB here adds complexity.